service/src/main/java/uk/nhs/adaptors/gp2gp/common/configuration/AppInitializer.java

```diff
@@ -32,13 +33,12 @@ public void afterPropertiesSet() {
         }
     }
 
-    @Bean
-    public S3Client getS3Client() {
+    public S3Client getS3Client() throws ConfigurationException {
         if (StringUtils.isNotBlank(storageConnectorConfiguration.getTrustStoreUrl())
             && storageConnectorConfiguration.getTrustStoreUrl().startsWith(S3_PREFIX)) {
             return S3Client.builder().build();
         }
 
-        return null;
+        throw new ConfigurationException("S3Client cannot be instantiated due to trust store URL misconfiguration");
```

Comment:
Any chance this error message can be made more specific? This is likely going to be shown in the error logs, and as an NME I'd have no idea how to fix this.

Reply:
thanks, I've adjusted the log message